File: G_Projects/G_Inventory_Management_System/ProductReplenishmentStrategies/ConcreteReplenishmentStrategies/BulkOrderStrategy.java
package G_Projects.G_Inventory_Management_System.ProductReplenishmentStrategies.ConcreteReplenishmentStrategies;

import G_Projects.G_Inventory_Management_System.ProductFactory.Product;
import G_Projects.G_Inventory_Management_System.ProductReplenishmentStrategies.ReplenishmentStrategy;

public class BulkOrderStrategy implements ReplenishmentStrategy {
    @Override
    public void replenish(Product product){
        System.out.println("Replenishing product " + product.getName() + " using Bulk Order Strategy.");
    }
}

File: Python/A_Project/D_Inventory_Management_System/ProductReplenishmentStrategies/ConcreteReplenishmentStrategies/bulk_order_strategy.py
from ProductReplenishmentStrategies.replenishment_strategy import ReplenishmentStrategy


class BulkOrderStrategy(ReplenishmentStrategy):
    def replenish(self, product):
        target_quantity = product.get_threshold() * 2
        if product.get_quantity() < target_quantity:
            product.set_quantity(target_quantity)
        print(
            f"Replenishing product {product.get_name()} using Bulk Order Strategy."
        )
